ambry-cloud/src/main/java/com/github/ambry/cloud/VcrReplicationManager.java

```diff
@@ -91,6 +112,30 @@ public void start() throws ReplicationException {
     vcrClusterParticipant.addListener(new VcrClusterParticipantListener() {
       @Override
       public void onPartitionAdded(PartitionId partitionId) {
+        if (partitionId.isEqual(cloudConfig.vcrHelixUpdaterParitionId)) {
+          vcrHelixUpdateLock.lock();
+          try {
+            if (!isAmbryListenerToUpdateVcrHelixRegistered) {
+              // Only register the listener once. Unfortunately, we can't unregister a listener, so we use
+              // isAmbryListenerToUpdateVcrHelixRegistered as the flag.
+              clusterMap.registerClusterMapListener(
+                  new com.github.ambry.cloud.VcrReplicationManager.AmbryListenerToUpdateVcrHelix());
+              isAmbryListenerToUpdateVcrHelixRegistered = true;
+              logger.info("VCR updater registered.");
+            }
+            // It's possible isVcrHelixUpdater to be true on two nodes.
+            // For example, lets say at time "t" node A is the owner of partition 1. Due to some partition reassignment
+            // (lets say new node addition), partition 1 get assigned to node B at time "t+1". In this case it's possible
```

Comment:
Let's say that we are trying to bootstrap a completely new vcr cluster after this change. Would we use the tool to create exactly one partition in that case and let this logic create all the other partitions?

Reply:
To bootstrap a VCR helix, my plan is to use previous CLI. Because setup up a helix involves too many configs. 

The scope of vcr automation is only to update VCR helix.

Reply:
We can discuss this offline. I was asking this with regards to automating the buildout of new datacenters. Helix is building a terraform provider for setting up new clusters programatically and I was trying to find out how far we are from being able to use that vs. making a new terraform step that would use the CLI tool we wrote.